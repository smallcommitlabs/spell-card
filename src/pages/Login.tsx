import React from 'react';
import { useForm } from 'react-hook-form';
import { useHistory } from 'react-router-dom';
import {
  Button,
  Card,
  CardBody,
  CardTitle,
  Col,
  Container,
  FormFeedback,
  FormGroup,
  Input,
  Label,
} from 'reactstrap';
import { AuthClient } from '../util/AuthClient';

interface formData {
  username: string;
  password: string;
}

const Login: React.FC = () => {
  const { register, handleSubmit, errors } = useForm();

  const onSubmit = ({ username, password }: formData) => {
    // This will become async once auth properly set up
    const res = AuthClient.signIn(username, password);
    if (res) {
      window.location.reload();
    }
  };

  return (
    <Container className={'min-vh-100 d-flex align-items-center'}>
      <Col sm={6} className={'col-sm-12 col-md-6 offset-md-3'}>
        <Card>
          <CardBody>
            <CardTitle tag='h1'>Login</CardTitle>
            <form onSubmit={handleSubmit(onSubmit)}>
              <FormGroup>
                <Label>Username</Label>
                <Input
                  name='username'
                  placeholder='Enter your username'
                  innerRef={register({
                    required: true,
                  })}
                  invalid={errors.username}
                />
                {errors.username && <FormFeedback>Please enter a username</FormFeedback>}
              </FormGroup>
              <FormGroup>
                <Label>Password</Label>
                <Input
                  type={'password'}
                  name='password'
                  placeholder='Enter your password'
                  innerRef={register({
                    required: true,
                  })}
                  invalid={errors.password}
                />
                {errors.password && <FormFeedback>Please enter a password</FormFeedback>}
              </FormGroup>
              <FormGroup>
                <Button type='submit' color={'primary'} block>
                  Login
                </Button>
              </FormGroup>
            </form>
          </CardBody>
        </Card>
      </Col>
    </Container>
  );
};

export default Login;
