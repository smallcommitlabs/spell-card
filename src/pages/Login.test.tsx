import { render } from '@testing-library/react';
import React from 'react';
import Login from './Login';

test('Page should match snapshot', () => {
  const tree = render(<Login />).asFragment();

  expect(tree).toMatchSnapshot();
});
