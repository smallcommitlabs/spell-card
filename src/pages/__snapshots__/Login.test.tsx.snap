// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="min-vh-100 d-flex align-items-center container"
  >
    <div
      class="col-sm-12 col-md-6 offset-md-3 col-sm-6"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <h1
            class="card-title"
          >
            Login
          </h1>
          <form>
            <div
              class="form-group"
            >
              <label
                class=""
              >
                Username
              </label>
              <input
                class="form-control"
                name="username"
                placeholder="Enter your username"
                type="text"
              />
            </div>
            <div
              class="form-group"
            >
              <label
                class=""
              >
                Password
              </label>
              <input
                class="form-control"
                name="password"
                placeholder="Enter your password"
                type="password"
              />
            </div>
            <div
              class="form-group"
            >
              <button
                class="btn btn-primary btn-block"
                type="submit"
              >
                Login
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
